### ISSUE TITLE
Shooting stars Spotify effect does nothing

### ISSUE THREAD

**Issue by The-Mighty-Spoon** [rowak/nanoleaf-desktop#29]
As the title says.
All the other effects seem to work fine-ish but as soon as i select the shooting star effect it does nothing no matter how many times i restart it

**Comment by rowak** [comment]
I've noticed this effect misbehaving lately as well. The streaking notes (shooting stars) effect has been broken for me ever since I rearranged my panel layout. I took a quick look into this and found that the problem is caused by a miscalculation. I fixed the problem for *my* layout, but it may behave differently on your layout, depending on how many "end" panels your layout has. An end panel is defined as any panel that is only connected to one other panel. End panels are where the streaks "enter" and "exit". In hindsight, this also means that if your layout has no end panels then this effect will not work for you.

Also note that the Nanoleaf Canvas will almost definitely **not** work with this effect because no code is written to support it.

[Here's a link](https://ufile.io/phh0ic2k) to a debug build with a possible fix. Let me know if it works for you. (note that if you on windows or mac, .jar files can be run the same as .exe files).

**Comment by The-Mighty-Spoon** [comment]
I seem to be unable to launch the .jar file, just get shown a "JNI error"
I should have mentioned that I'm using the Canvas Light panels, sorry about that

**Comment by rowak** [comment]
Yeah sorry this effect won't work with the Canvas. Also, not that it matters, but I think the JNI error is caused by having an outdated version of Java installed. NFD is usually compiled with Java 8, but I recently installed to Java 11 on my system so that .jar was probably compiled with Java 11.

I'm going to leave this issue open until I release a new update as a reminder to myself to make sure this issue is fixed for Aurora devices.
